core/src/main/java/org/elasticsearch/script/Script.java

```diff
@@ -63,25 +66,27 @@ public Script(String script) {
      * For sub-classes to use to override the default language
      */
     protected Script(String script, String lang) {
-        this(script, ScriptType.INLINE, lang, null);
+        this(script, ScriptType.INLINE, lang, null, null);
+    }
+
+    public Script(String script, ScriptType type, @Nullable String lang, @Nullable Map<String, ?> params) {
+        this(script, type, lang, params, null);
     }
 
     /**
      * Constructor for Script.
      *
-     * @param script
-     *            The cache key of the script to be compiled/executed. For
-     *            inline scripts this is the actual script source code. For
-     *            indexed scripts this is the id used in the request. For on
-     *            file scripts this is the file name.
-     * @param type
-     *            The type of script -- dynamic, indexed, or file.
-     * @param lang
-     *            The language of the script to be compiled/executed.
-     * @param params
-     *            The map of parameters the script will be executed with.
+     * @param script        The cache key of the script to be compiled/executed. For
+     *                      inline scripts this is the actual script source code. For
+     *                      indexed scripts this is the id used in the request. For on
+     *                      file scripts this is the file name.
+     * @param type          The type of script -- dynamic, indexed, or file.
+     * @param lang          The language of the script to be compiled/executed.
+     * @param params        The map of parameters the script will be executed with.
+     * @param contentType   The {@link XContentType} of the script.
```

Comment:
I think this needs a bit more explanation, as well as what null means. Honestly it'd be best if we could avoid it being part of script altogether because most scripts don't have a content type.